cmd/revad/svcs/grpcsvcs/gatewaysvc/publicshareprovidersvc.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018-2019 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package gatewaysvc
+
+import (
+	"context"
+
+	publicshareproviderv0alphapb "github.com/cs3org/go-cs3apis/cs3/publicshareprovider/v0alpha"
+	rpcpb "github.com/cs3org/go-cs3apis/cs3/rpc"
+	"github.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/pool"
+	"github.com/cs3org/reva/cmd/revad/svcs/grpcsvcs/status"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/pkg/errors"
+)
+
+func (s *svc) CreatePublicShare(ctx context.Context, req *publicshareproviderv0alphapb.CreatePublicShareRequest) (*publicshareproviderv0alphapb.CreatePublicShareResponse, error) {
```

Comment:
@refs add todos to these functions to indicate that they need to implement the same dual commit strategy as the user share provider gateway implementation

Reply:
will add this changes to the next PR that integrates the public share provider into the `ocssvc` and keep this PR clean 👍

Reply:
ah well ... todos won't hurt? I don't want to loose track of it. annotate them with `//TODO(refs): ` ... Hugo likes to grep the codebase ... I personally use the [TODO Tree extension](https://github.com/Gruntfuggly/todo-tree) for VS Code.

Reply:
ouch, sorry, I meant to address the other comment 🤦‍♂